lychee-lib/src/extract.rs

```diff
@@ -155,20 +144,15 @@ pub(crate) fn extract_links(
     // Silently ignore the parse failures for now.
     let mut requests: HashSet<Request> = HashSet::new();
     for link in links {
-        match Uri::try_from(link.as_str()) {
-            Ok(uri) => {
-                requests.insert(Request::new(uri, input_content.input.clone()));
-            }
-            Err(_) => {
-                if !Path::new(&link).exists() {
-                    if let Some(base_url) = &base_url {
-                        if let Ok(new_url) = base_url.join(&link) {
-                            requests.insert(Request::new(
-                                Uri::Website(new_url),
-                                input_content.input.clone(),
-                            ));
-                        }
-                    }
+        if let Ok(uri) = Uri::try_from(link.as_str()) {
```

Comment:
Maybe we should move that block into its own method. The nested if lets are not the easiest to read. We can do that at a future point in time, though.

Reply:
I can combine the two `if let` to one
```rust
            if let Some(new_url) = base_url.as_ref().and_then(|u| u.join(&link).ok()) {
                requests.insert(Request::new(
                    Uri::Website(new_url),
                    input_content.input.clone(),
                ));
            }
```

Reply:
Looks good